src/app/outlays/edit-outlay-dialog/edit-outlay-dialog.component.ts

```diff
@@ -0,0 +1,390 @@
+  ) {
+    dialogRef.disableClose = true;
+  }
+
+  ngOnInit() {
+    this.dontwantData = this.data.dontwantData;
+    this.isDispatched = this.dontwantData.dispatched;
+    this.user = this.userService.getUser();
+    console.warn('data', this.data);
+
+    this.editDontwantForm = this.fb.group({
+      technicalStuff: [
+        {
+          technicalStuff: this.dontwantData.technicalStuff || null,
+          type: this.dontwantData.type || null,
+          otherTechnicalStuff: this.dontwantData.otherTechnicalStuff || null,
+        },
+        [ Validators.required ],
+      ],
+      hsDispatchDate: [
+        {
+          value: this.dontwantData.hsDispatchDate
+            ? this.dateService.getValidDateStringFormatFromDate(this.dontwantData.hsDispatchDate)
+            : null,
+          disabled: this.user.isTechnician,
+        },
+        [ this.dateService.validateDate(this.dateService) ]
+      ],
+      hsDispatchDateTechnician: [
+        this.dontwantData.hsDispatchDateTechnician
+          ? this.dateService.getValidDateStringFormatFromDate(this.dontwantData.hsDispatchDateTechnician)
+          : null,
+        [ this.dateService.validateDate(this.dateService) ],
+      ],
+      createdAt: [
+        this.dontwantData.createdAt
+          ? this.dateService.getValidDateStringFormatFromDate(this.dontwantData.createdAt)
+          : this.dateService.getCurrentDateAsString(),
+        [ this.dateService.validateDate(this.dateService) ],
+      ],
+      assignedTo: [
+        this.dontwantData.assignedTo ? this.dontwantData.assignedTo.uid : null,
+      ],
+      customerName: [
+        this.dontwantData.customerName,
+        [ Validators.required ],
+      ],
+      customerPhone: [
+        this.dontwantData.customerPhone,
+        [ Validators.required, Validators.pattern('^[9]{1}[0-9]{2}[ ]{0,1}[0-9]{3}[ ]{0,1}[0-9]{3}$')],
+      ],
+      note: [ this.dontwantData.note || null ],
+    });
+
+    this.manageDispatchEligibility();
+
+    if (this.dontwantData.hsDispatchDate) {
+      this.editDontwantForm.disable();
+    }
+
+    this.editDontwantFormChanges$ = this.editDontwantForm.valueChanges.subscribe((value: string) => {
+      this.manageDispatchEligibility();
+    });
+
+    this.getTechnicians();
+  }
+
+  get technicalStuff() { return this.editDontwantForm.get('technicalStuff'); }
+  get type() { return this.editDontwantForm.get('type'); }
+  get customerName() { return this.editDontwantForm.get('customerName'); }
+  get customerPhone() { return this.editDontwantForm.get('customerPhone'); }
+  get hsDispatchDate() { return this.editDontwantForm.get('hsDispatchDate'); }
+  get hsDispatchDateTechnician() { return this.editDontwantForm.get('hsDispatchDateTechnician'); }
+
+  getFormValidationErrors() {
+    this.editDontwantForm.markAllAsTouched();
+    this.inputtechnicalStuffGroups.forEach(comp => {
+      comp.technicalStuffForm.markAllAsTouched();
+    });
+  }
+
+  getTechnicians() {
```

Comment:
nit: I always wait a method with prefix "get" returns something, such method looks more like initTechnicans$ for me